compiler/src/iree/compiler/Codegen/Common/GPU/WorkgroupReordering.cpp

```diff
@@ -68,6 +68,95 @@ makeSwizzledIds(Location loc, OpBuilder b, Value workgroupIdX,
   return {swizzledIdX, swizzledIdY};
 }
 
+// reoredering to make workgroup ids move slowly between chiplet groups
+static Value chipletAwareWorkgroupReordering(Location loc, OpBuilder b,
+                                             Value linearizedId,
+                                             Value workgroupCountX,
+                                             Value workgroupCountY,
+                                             int64_t numChipletsPerGroup) {
```

Comment:
Is this the number of chiplets per *work*group or something else?

Reply:
The number of partitions on GPU, value is typically either 4, with two XCDs per partition, or 8, with one XCD per partition. 

here I use the same naming convention as RocMLIR, which is misleading. I will adopt a more accurate name to better reflect its purpose.